### PR TITLE
Restore TTY when aborting TOTP ticker with Ctrl+C

### LINKED ISSUE #2320: gopass totp disabled terminal echo
### Summary
After running `gopass totp ...`, and ctrl-c to exit, terminal input is hidden, because echo was disabled.

### Steps To Reproduce
- Run `gopass totp ...`
- Use Ctrl-C to exit
- Try to type something
- Input is not displayed on the terminal

Note:
- If `q` is used to exit, then the terminal input IS correctly displayed.

### Expected behavior
- Terminal input should be displayed in both cases. 
- Terminal echo should be re-enabled as soon as gopass has decrypted the secret, long before doing anything else.

### Environment
- OS: Gentoo Linux
- OS version: `Linux REDACTED 5.18.15-gentoo-dist #1 SMP PREEMPT_DYNAMIC Fri Jul 29 22:03:23 -00 2022 x86_64 Intel(R) Core(TM) REDACTED GenuineIntel GNU/Linux`
- gopass Version: 1.14.4
- Installation method: distro package manager

### PR DESCRIPTION
**Pull request by dominikschulz**
Fixes #2320

RELEASE_NOTES=[BUGFIX] Handle Ctrl+C in TOTP

Signed-off-by: Dominik Schulz <dominik.schulz@gauner.org>

### CODE REVIEW

**Comment by dominikschulz** [comment]
This makes an tricky code path even more tricky :/

**Comment by dominikschulz** [comment]
The failing test looks like a flake to me.

**Comment by dominikschulz** [comment]
No, it's an actual concurrency issue.

File: main.go [hunk 1]
```diff
@@ -70,6 +70,7 @@ func main() {
 	go func() {
 		select {
 		case <-sigChan:
+			fmt.Println("SIGCHAN")
```
Comment (AnomalRoil) [inline main.go:73]: Is this really meant to be left in prod code? Maybe make it a debug level log?

File: internal/action/otp.go [hunk 2]
```diff
@@ -97,7 +97,9 @@ func (s *Action) otp(ctx context.Context, name, qrf string, clip, pw, recurse bo
 	skip := ctxutil.IsHidden(ctx) || pw || qrf != "" || !ctxutil.IsTerminal(ctx) || !ctxutil.IsInteractive(ctx) || clip
 	if !skip {
 		// let us monitor key presses for cancellation:.
-		go waitForKeyPress(ctx, cancel)
+		runFn, cleanupFn := waitForKeyPress(ctx, cancel)
+		go runFn()
+		defer cleanupFn()
```
Comment (AnomalRoil) [inline internal/action/otp.go:102]: Good catch!

**Review by AnomalRoil — DISMISSED** [review]

File: main.go (hunk 1 above)
Comment (dominikschulz) [reply main.go:73]: Ohmy. No, this should go away of course. Thanks :)

**Pull request merged**